engine/interfaces.go

```diff
@@ -31,6 +33,7 @@ type ImmediateTaskQueue interface {
 	// TriggerPollingTasks exposes an API to be called by TaskNotifier
 	TriggerPollingTasks(request xcapi.NotifyImmediateTasksRequest)
 	Stop(ctx context.Context) error
+	UpdateShardId(shardId int32)
```

Comment:
I don't follow this, why we need to update shardId? We can sync offline

Reply:
It's used to override the shardId so we can combile the operation of destroy + create shards into updateShardId.

Reply:
Deleted this API since we don't do the updation.